src/um/fltuser.rs

```diff
@@ -0,0 +1,376 @@
+    VolumeFileSystemType: FLT_FILESYSTEM_TYPE,
+    InstanceNameLength: USHORT,
+    InstanceNameBufferOffset: USHORT,
+    AltitudeLength: USHORT,
+    AltitudeBufferOffset: USHORT,
+    VolumeNameLength: USHORT,
+    VolumeNameBufferOffset: USHORT,
+    FilterNameLength: USHORT,
+    FilterNameBufferOffset: USHORT,
+    SupportedFeatures: ULONG,
+}}
+STRUCT!{struct INSTANCE_AGGREGATE_STANDARD_INFORMATION_u_s_LegacyFilter {
+    Flags: ULONG,
+    AltitudeLength: USHORT,
+    AltitudeBufferOffset: USHORT,
+    VolumeNameLength: USHORT,
+    VolumeNameBufferOffset: USHORT,
+    FilterNameLength: USHORT,
+    FilterNameBufferOffset: USHORT,
+    SupportedFeatures: ULONG,
+}}
+UNION!{union INSTANCE_AGGREGATE_STANDARD_INFORMATION_Type_u {
+    [u32; 8],
+    MiniFilter mut_MiniFilter: INSTANCE_AGGREGATE_STANDARD_INFORMATION_u_s_MiniFilter,
+    LegacyFilter mut_LegacyFilter: INSTANCE_AGGREGATE_STANDARD_INFORMATION_u_s_LegacyFilter,
+}}
+STRUCT!{struct INSTANCE_AGGREGATE_STANDARD_INFORMATION {
+    NextEntryOffset: ULONG,
+    Flags: ULONG,
+    Type: INSTANCE_AGGREGATE_STANDARD_INFORMATION_Type_u,
+}}
+STRUCT!{struct FILTER_MESSAGE_HEADER {
+    ReplyLength: ULONG,
+    MessageId: ULONGLONG,
+}}
+STRUCT!{struct FILTER_REPLY_HEADER {
+    Status: NTSTATUS,
+    MessageId: ULONGLONG,
+}}
+pub type PFILTER_MESSAGE_HEADER = *mut FILTER_MESSAGE_HEADER;
+pub type PFILTER_REPLY_HEADER = *mut FILTER_REPLY_HEADER;
+pub type PFILTER_FULL_INFORMATION = *mut FILTER_FULL_INFORMATION;
+pub type PFILTER_AGGREGATE_BASIC_INFORMATION = *mut FILTER_AGGREGATE_BASIC_INFORMATION;
+pub type PFILTER_AGGREGATE_STANDARD_INFORMATION = *mut FILTER_AGGREGATE_STANDARD_INFORMATION;
+pub type PFILTER_VOLUME_BASIC_INFORMATION = *mut FILTER_VOLUME_BASIC_INFORMATION;
+pub type PFILTER_VOLUME_STANDARD_INFORMATION = *mut FILTER_VOLUME_STANDARD_INFORMATION;
+pub type PINSTANCE_BASIC_INFORMATION = *mut INSTANCE_BASIC_INFORMATION;
+pub type PINSTANCE_PARTIAL_INFORMATION = *mut INSTANCE_PARTIAL_INFORMATION;
+pub type PINSTANCE_FULL_INFORMATION = *mut INSTANCE_FULL_INFORMATION;
+pub type PINSTANCE_AGGREGATE_STANDARD_INFORMATION = *mut INSTANCE_AGGREGATE_STANDARD_INFORMATION;
+extern "system" {
+    pub fn FilterAttach(
+        lpFilterName: LPCWSTR,
+        lpVolumeName: LPCWSTR,
+        lpInstanceName: LPCWSTR,
+        dwCreatedInstanceNameLength: DWORD,
+        lpCreatedInstanceName: LPWSTR
+    ) -> HRESULT;
+    pub fn FilterAttachAtAltitude(
+        lpFilterName: LPCWSTR,
+        lpVolumeName: LPCWSTR,
+        lpAltitude: LPCWSTR,
+        lpInstanceName: LPCWSTR,
+        dwCreatedInstanceNameLength: DWORD,
+        lpCreatedInstanceName: LPWSTR
+    ) -> HRESULT;
+    pub fn FilterClose(
+        hFilter: HFILTER
+    ) -> HRESULT;
+    pub fn FilterConnectCommunicationPort(
+        lpPortName: LPCWSTR,
+        dwOptions: DWORD,
+        lpContext: LPCVOID,
+        wSizeOfContext: WORD,
+        lpSecurityAttributes: LPSECURITY_ATTRIBUTES,
+        hPort: PHANDLE
```

Comment:
```suggestion
        hPort: *mut HANDLE
```

The windows header file uses a raw pointer here.